### PR TITLE
chore: run all unit tests on CI

### PR DESCRIPTION
**Pull request by terakoya76**
Part of https://github.com/SigNoz/signoz/issues/236

I found a unit test that was degraded by https://github.com/SigNoz/signoz/pull/2036. 

I think that this occurred because CI did not run all unit tests, so I would like to make sure that they do.
I hope this will prevent the unit tests that we will be adding from now on from being degraded.

I tested gh-actions workflow works on my forked repo's PR
* https://github.com/terakoya76/signoz/pull/2
* https://github.com/terakoya76/signoz/actions/runs/4210018578/jobs/7307402587

### CODE REVIEW

File: deploy/docker/clickhouse-setup/docker-compose-core.yaml [hunk 1]
```diff
@@ -30,9 +30,9 @@ services:
     image: signoz/alertmanager:0.23.0-0.2
     volumes:
       - ./data/alertmanager:/data
-    depends_on:
-      query-service:
-        condition: service_healthy
+    # depends_on:
```
Comment (terakoya76) [inline deploy/docker/clickhouse-setup/docker-compose-core.yaml:33]: Disable dependency on query-service container to allow docker-compose-core to run alone

File: pkg/query-service/app/http_handler_test.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func TestPrepareQuery(t *testing.T) {
 				Query: "ALTER TABLE signoz_table DELETE where true",
 			},
 			expectedErr: true,
-			errMsg:      "query shouldn't alter data",
```
Comment (terakoya76) [inline pkg/query-service/app/http_handler_test.go:70]: this failed like this
```
$ make test
# go.signoz.io/signoz/pkg/query-service/app/clickhouseReader
pkg/query-service/app/clickhouseReader/options_test.go:68:26: missing ',' before newline in composite literal
FAIL    go.signoz.io/signoz/pkg/query-service/app/clickhouseReader [setup failed]
?       go.signoz.io/signoz/pkg/query-service   [no test files]
--- FAIL: TestPrepareQuery (0.00s)
    --- FAIL: TestPrepareQuery/query_contains_alter_table (0.00s)
        http_handler_test.go:120: expected error message contain: query shouldn't alter data, but got: Operation alter table is not allowed
.
1 total assertion

..
3 total assertions

.
4 total assertions

FAIL
coverage: 1.4% of statements
FAIL    go.signoz.io/signoz/pkg/query-service/app       0.028s
?       go.signoz.io/signoz/pkg/query-service/app/dashboards    [no test files]
ok      go.signoz.io/signoz/pkg/query-service/app/logs  (cached)        coverage: 53.9% of statements
ok      go.signoz.io/signoz/pkg/query-service/app/metrics       (cached)        coverage: 62.7% of statements
?       go.signoz.io/signoz/pkg/query-service/app/parser        [no test files]
?       go.signoz.io/signoz/pkg/query-service/auth      [no test files]
ok      go.signoz.io/signoz/pkg/query-service/constants (cached)        coverage: 63.6% of statements
?       go.signoz.io/signoz/pkg/query-service/dao       [no test files]
?       go.signoz.io/signoz/pkg/query-service/dao/sqlite        [no test files]
?       go.signoz.io/signoz/pkg/query-service/featureManager    [no test files]
?       go.signoz.io/signoz/pkg/query-service/healthcheck       [no test files]
?       go.signoz.io/signoz/pkg/query-service/integrations/alertManager [no test files]
?       go.signoz.io/signoz/pkg/query-service/integrations/signozio     [no test files]
?       go.signoz.io/signoz/pkg/query-service/interfaces        [no test files]
?       go.signoz.io/signoz/pkg/query-service/model     [no test files]
?       go.signoz.io/signoz/pkg/query-service/pqlEngine [no test files]
?       go.signoz.io/signoz/pkg/query-service/rules     [no test files]
?       go.signoz.io/signoz/pkg/query-service/telemetry [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils     [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/encryption  [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/labels      [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/queryTemplate       [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/times       [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/timestamp   [no test files]
?       go.signoz.io/signoz/pkg/query-service/utils/value       [no test files]
?       go.signoz.io/signoz/pkg/query-service/version   [no test files]
FAIL
```

File: Makefile [hunk 3]
```diff
@@ -119,6 +119,16 @@ down-local:
 	$(STANDALONE_DIRECTORY)/docker-compose-core.yaml -f $(STANDALONE_DIRECTORY)/docker-compose-local.yaml \
 	down -v
 
+run-dependencies:
+	@LOCAL_GOOS=$(LOCAL_GOOS) LOCAL_GOARCH=$(LOCAL_GOARCH) docker-compose -f \
+	$(STANDALONE_DIRECTORY)/docker-compose-core.yaml \
+	up --build -d
```
Comment (prashant-shahi) [inline Makefile:125]: No need to pass `LOCAL_*` variables as well as `--build` flag.
They are used by `docker-compose-local.yaml` to build docker images with local changes.

```suggestion
	@docker-compose -f $(STANDALONE_DIRECTORY)/docker-compose-core.yaml up -d
```

File: Makefile [hunk 4]
```diff
@@ -137,4 +147,4 @@ clear-swarm-data:
 	sh -c "cd /pwd && rm -rf alertmanager/* clickhouse*/* signoz/* zookeeper-*/*"
 
 test:
-	go test ./pkg/query-service/app/metrics/...
+	@SIGNOZ_LOCAL_DB_PATH="$(CURDIR)/pkg/query-service/signoz.db" go test -cover $(shell go list ./pkg/query-service/... | grep -v tests)
```
Comment (prashant-shahi) [inline Makefile:150]: @srikanthccv @makeavish can anyone of you guys review this section?

File: deploy/docker/clickhouse-setup/docker-compose-core.yaml (hunk 1 above)
Comment (prashant-shahi) [reply deploy/docker/clickhouse-setup/docker-compose-core.yaml:33]: makes sense :+1:

**Review by prashant-shahi — CHANGES_REQUESTED** [review]
left few comments.

Also, asked @srikanthccv @makeavish to have a look at test command

**Comment by srikanthccv** [comment]
No longer relevant.

**Pull request closed without merging**